pygmt/tests/test_grdview.py

```diff
@@ -57,199 +69,204 @@ def test_grdview_wrong_kind_of_grid(grid):
+def test_grdview_with_cmap_for_perspective_surface_plot(gridfile, grid):
     """
     Run grdview by passing in a grid and setting a colormap for producing a
     surface plot with a 3D perspective viewpoint.
     """
-    fig = Figure()
-    fig.grdview(
+    fig_ref = Figure()
+    fig_ref.grdview(
+        grid=gridfile, cmap="oleron", surftype="s", perspective=[225, 30], zscale=0.005
+    )
+    fig_test = Figure()
+    fig_test.grdview(
         grid=grid, cmap="oleron", surftype="s", perspective=[225, 30], zscale=0.005
     )
-    return fig
+    return fig_ref, fig_test
 
 
-@pytest.mark.xfail(
-    reason="Baseline image not updated to use earth relief grid in GMT 6.1.0",
-)
-@pytest.mark.mpl_image_compare
-def test_grdview_on_a_plane(grid):
+@check_figures_equal()
+def test_grdview_on_a_plane(gridfile, grid):
     """
     Run grdview by passing in a grid and plotting it on a z-plane, while
     setting a 3D perspective viewpoint.
     """
-    fig = Figure()
-    fig.grdview(grid=grid, plane=-4000, perspective=[225, 30], zscale=0.005)
-    return fig
+    fig_ref = Figure()
+    fig_ref.grdview(grid=gridfile, plane=-4000, perspective=[225, 30], zscale=0.005)
+    fig_test = Figure()
+    fig_test.grdview(grid=grid, plane=-4000, perspective=[225, 30], zscale=0.005)
+    return fig_ref, fig_test
 
 
-@pytest.mark.xfail(
-    reason="Baseline image not updated to use earth relief grid in GMT 6.1.0",
-)
-@pytest.mark.mpl_image_compare
-def test_grdview_on_a_plane_with_colored_frontal_facade(grid):
+@check_figures_equal()
+def test_grdview_on_a_plane_with_colored_frontal_facade(gridfile, grid):
     """
     Run grdview by passing in a grid and plotting it on a z-plane whose frontal
     facade is colored gray, while setting a 3D perspective viewpoint.
     """
-    fig = Figure()
-    fig.grdview(grid=grid, plane="-4000+ggray", perspective=[225, 30], zscale=0.005)
-    return fig
+    fig_ref = Figure()
+    fig_ref.grdview(
+        grid=gridfile, plane="-4000+ggray", perspective=[225, 30], zscale=0.005
+    )
+    fig_test = Figure()
+    fig_test.grdview(
+        grid=grid, plane="-4000+ggray", perspective=[225, 30], zscale=0.005
+    )
+    return fig_ref, fig_test
 
 
-@pytest.mark.xfail(
-    reason="Baseline image not updated to use earth relief grid in GMT 6.1.0",
-)
-@pytest.mark.mpl_image_compare
-def test_grdview_with_perspective_and_zaxis_frame(grid):
+@check_figures_equal()
+def test_grdview_with_perspective_and_zaxis_frame(gridfile, grid):
     """
     Run grdview by passing in a grid and plotting an annotated vertical
     z-axis frame.
     """
-    fig = Figure()
-    fig.grdview(grid=grid, perspective=[225, 30], zscale=0.005, frame="zaf")
-    return fig
+    fig_ref = Figure()
+    fig_ref.grdview(grid=gridfile, perspective=[225, 30], zscale=0.005, frame="zaf")
+    fig_test = Figure()
+    fig_test.grdview(grid=grid, perspective=[225, 30], zscale=0.005, frame="zaf")
+    return fig_ref, fig_test
```

Comment:
This ` test_grdview_with_perspective_and_zaxis_frame` is a very strange one. It passes when the full test suite is ran using `make test`, but was failing locally for me everytime I ran `pytest --mpl --verbose --doctest-modules pygmt/tests/test_grdview.py`. Even weirder is that the problem seems to be on the NetCDF plot rather than the xarray one (now I've seen everything)! The difference is a doubling on the x and y-axis frame, will report this to upstream GMT in a bit:

| NetCDF (expected, but wrong) | xarray.DataArray (test, correct?!!) |
|---|---|
| ![test_grdview_with_perspective_and_zaxis_frame-expected](https://user-images.githubusercontent.com/23487320/92370449-6a37b400-f14e-11ea-968f-6a520bb7d6aa.png)| ![test_grdview_with_perspective_and_zaxis_frame](https://user-images.githubusercontent.com/23487320/92370444-69068700-f14e-11ea-8717-f7194f26bb58.png) |

Exact error is:
```
matplotlib.testing.exceptions.ImageComparisonFailure: Image sizes do not match expected size: (1534, 2664, 3) actual size (1521, 2638, 3)
```

Reply:
The xarray.DataArray result is "incorrect", because GMT takes the input xarray grid as a Cartesian grid, although `grid.gmt.gtype=1`.

Reply:
Ok, so there's several layers of 'wrong-ness' here it seems? The NetCDF grdview plot is wrong because it has a double frame (without the zebra stripes), and the xarray grid is wrong because it's using the Cartesian plain map frame? But these are 'absolute' wrongs i.e. bugs, and our new `@check_figures_equal` decorator only does a 'relative' comparison to ensure we're reproducing GMT plots in PyGMT (be it right or wrong).

What I don't understand is why the tests pass when ran using `make test` (false negative), but fails with `pytest --mpl --verbose --doctest-modules pygmt/tests/test_grdview.py`. Seems like another flaky test issue.

Reply:
Yes, it puzzles me, too.